### PR TITLE
feat(composition): add composition for gke

### PR DESCRIPTION
**Pull request by haarchri**
### Description of your changes
- add composition
- add example
- clean makefile
- add more providers in crossplane.yaml

Fixes #

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

```
--- PASS: kuttl (1255.05s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/case (1254.54s)
PASS
18:48:42 [ OK ] running automated tests
```

### CODE REVIEW

**Review by humoflife — COMMENTED** [review]
Should we use Crossplane 1.14.5 in crossplane.yaml?

File: apis/definition.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                      enum:
+                        - Delete
+                        - Orphan
+                      type: string
+                      default: Delete
+                    providerConfigName:
+                      description: Crossplane ProviderConfig to use for provisioning this resources
+                      type: string
+                      default: default
+                    version:
```
Comment (ytsarev) [inline apis/definition.yaml:44]: it looks like this field is not used anywhere in the Composition?

Comment (haarchri) [reply apis/definition.yaml:44]: thanks i missed 2 patches :(

**Review by ytsarev — APPROVED** [review]
Thanks for the fixes, LGTM 👍

**Pull request merged**